### ISSUE TITLE
feat: allow using logos images from the template directory

### ISSUE THREAD

**Issue by matifali** [coder/coder#6339]
## Problem
We can not use logos/images in the template directory.
Setting `icon = ./my-logo.svg` does not seem to work.
## Workaround
I can upload it to a CDN, and it works, but it's easier if we can include the logos with the template. More intuitive and will also work for the air-gapped scenario.

cc: @BrunoQuaresma

**Comment by sharkymark** [comment]
2nd this x 10 @matifali

**Comment by MrPeacockNLB** [comment]
Maybe not just icons. Currently we have to setup a Nginx to serve a schema.json to do a validation for a yaml which gets parsed inside the init-script. So pushing not only one file and not only a specific type of files.

**Comment by BrunoQuaresma** [comment]
I think the broad idea would have a way to upload static files together so it could be useful for more scenarios.

**Comment by deansheather** [comment]
Right now we just store the tar file that gets uploaded by the client, so to get a single file out of the tar we would have to query the entire tar from the DB into memory, scan it, and then serve the file, which would be very efficient in a hot path (like showing the image on the workspaces page).

We may need to come up with a different way of storing files for templates to support something like this, or just allow uploading files separately perhaps.

**Comment by ammario** [comment]
We could accept the tar with a Tee reader, and if we see a certain file (e.g. "icon.[png|svg|jpeg....|"), store the icon in a separate table. It would be unfortunate to have two different ways of setting the template icon.

**Comment by bpmct** [comment]
> We could accept the tar with a Tee reader, and if we see a certain file (e.g. "icon.[png|svg|jpeg....|"), store the icon in a separate file. It would be unfortunate to have two different ways of setting the template icon.

The README.md can actually already be used to set a default icon for templates, but after that it is managed by the UI/CLI/database (not the template state).

**Comment by matifali** [comment]
This is not about just the icon of the template but all the icons we use for `coder_parameter` data

**Comment by mtojek** [comment]
Let's park it as unassigned for now.

**Comment by MrPeacockNLB** [comment]
please do not close this issue

**Comment by matifali** [comment]
@johnhorsema Would this be possible as part of https://github.com/coder/coder/issues/13007?

**Comment by johnhorsema** [comment]
> @johnhorsema Would this be possible as part of #13007? I think we may be able to set icons but would still need them hosted somewhere.

I think you tagged the wrong person, should be @johnstcn?

**Comment by ammario** [comment]
@matifali if we parse out the icon from the template archive then the feature shouldn't affect #13007. If we accept a direct file upload (which I don't see a good reason to do) we will have to change #13007 as well as the CLI and UI.

**Comment by matifali** [comment]
Yes parsing for icons from the archive could be an option. It's still something nice to have so should not block work ok any other features.

**Comment by BrunoQuaresma** [comment]
If we are ok with the @johnstcn solution, I would like to take it on the next sprint.
